### ISSUE TITLE
Unable to login: "failed to build login form data: could not find any forms matching the provided IDs"

### ISSUE THREAD

**Issue by bonnici** [Versent/saml2aws#811]
I can't assume into a role using my normal login command:
`saml2aws login --role=arn:aws:iam::xxx:role/xx`

The output I get is:
```
Authenticating as [xxx@xxx.com](xxx@xxx.com) ...
error authenticating to IdP: error loading login page: failed to build login form data: could not find any forms matching the provided IDs
```

I have tried using the `--force` option but get the same error. I also tried logging in in an incognito browser window and that worked, so I know it's not a password/MFA issue. When I try dumping the content `DUMP_CONTENT=true saml2aws login --verbose`, I get a HTML page with the error "The browser you’re using doesn’t support JavaScript, or has JavaScript turned off."

![Screen Shot 2022-04-27 at 9 26 32 am](https://user-images.githubusercontent.com/1110099/165409205-5e7eb884-a5a0-4303-9cd3-ce1edb0ea3c1.png)

I've tried using versions 2.27.1 and 2.34.0 (which is the latest on brew).

**Comment by miseyu** [comment]
Here, I had a similar situation. :bow:

```
Error authenticating to IdP.: error loading login page: failed to build login form data: could not find any forms matching the provided IDs
```

**Comment by apatnaik14** [comment]
Changing the provider in the ~/.saml2aws configuration to `Browser` seems to work. I was able to get that working locally, it launches a Chromium instance where you provide the credentials and that seems to complete the login.

But on my vms where I do not have a gui enabled, I am unable to get that working. 

`ERROR:browser_main_loop.cc(1400)] Unable to open X display.`

I saw some information around setting the `bgresponse` to js_enabled which solves the issue as well, but so far I haven't figured out where that setting is expected to be set. I would really appreciate it if anyone finds some instructions around it, if they could share it here.

**Comment by miseyu** [comment]
> Changing the provider in the ~/.saml2aws configuration to Browser seems to work. I was able to get that working locally, it launches a Chromium instance where you provide the credentials and that seems to complete the login.

@apatnaik14 
I tried this and was able to log in successfully. Thanks for the information. 🙇

**Comment by bonnici** [comment]
> Changing the provider in the ~/.saml2aws configuration to `Browser` seems to work. I was able to get that working locally, it launches a Chromium instance where you provide the credentials and that seems to complete the login.

Thanks! Didn't even know that was an option.

**Comment by jmctune** [comment]
> I saw some information around setting the bgresponse to js_enabled which solves the issue as well, but so far I haven't figured out where that setting is expected to be set. I would really appreciate it if anyone finds some instructions around it, if they could share it here.

You need to build the PR yourself from source and use the produced binary to get the fix. It's a code change, not a config change.

https://github.com/Versent/saml2aws#building

**Comment by apatnaik14** [comment]
> > I saw some information around setting the bgresponse to js_enabled which solves the issue as well, but so far I haven't figured out where that setting is expected to be set. I would really appreciate it if anyone finds some instructions around it, if they could share it here.
> 
> You need to build the PR yourself from source and use the produced binary to get the fix. It's a code change, not a config change.
> 
> https://github.com/Versent/saml2aws#building

Thanks @jmctune ! I was able to figure out the build process in the VM and got it working with the new change.

**Comment by luckielordie** [comment]
> Changing the provider in the ~/.saml2aws configuration to `Browser` seems to work. I was able to get that working locally, it launches a Chromium instance where you provide the credentials and that seems to complete the login.

This has worked for other people in our group, but If you're running from inside WSL it errors with `Unable to open X display`

`<user>@<machine_name>:~/git/deployments$ saml2aws login --disable-keychain
Using IdP Account default to access Browser <url>
To use saved password just hit enter.
? Username <email>
? Password ***********

Authenticating as <email> ...
Downloading browsers...
Downloaded browsers successfully
Error authenticating to IdP.: could not send message: could not send message to server: Protocol error (Browser.getVersion): Browser closed.
==================== Browser output: ====================
<launching> /home/<user>/.cache/ms-playwright/chromium-907428/chrome-linux/chrome --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=ImprovedCookieControls,LazyFrameLoading,GlobalMediaControls,DestroyProfileOnBrowserClose --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --disable-sync --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --no-service-autorun --user-data-dir=/tmp/playwright_chromiumdev_profile-HX3mZL --remote-debugging-pipe --no-sandbox --no-startup-window
<launched> pid=6535
[pid=6535][err] [6535:6535:0428/111120.308865:ERROR:browser_main_loop.cc(1400)] Unable to open X display.
=========================== logs ===========================
<launching> /home/mattl/.cache/ms-playwright/chromium-907428/chrome-linux/chrome --disable-background-networking --enable-features=NetworkService,NetworkServiceInProcess --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-breakpad --disable-client-side-phishing-detection --disable-component-extensions-with-background-pages --disable-default-apps --disable-dev-shm-usage --disable-extensions --disable-features=ImprovedCookieControls,LazyFrameLoading,GlobalMediaControls,DestroyProfileOnBrowserClose --allow-pre-commit-input --disable-hang-monitor --disable-ipc-flooding-protection --disable-popup-blocking --disable-prompt-on-repost --disable-renderer-backgrounding --disable-sync --force-color-profile=srgb --metrics-recording-only --no-first-run --enable-automation --password-store=basic --use-mock-keychain --no-service-autorun --user-data-dir=/tmp/playwright_chromiumdev_profile-HX3mZL --remote-debugging-pipe --no-sandbox --no-startup-window
<launched> pid=6535
[pid=6535][err] [6535:6535:0428/111120.308865:ERROR:browser_main_loop.cc(1400)] Unable to open X display.
============================================================`

**Comment by bonnici** [comment]
Can confirm this is fixed in v2.35
